### ISSUE TITLE
Tree path length

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tskit#2249]
Define a function that computes the number of edges traversed from one node to another in a tree.

I think the simplest way to define this is in terms of the depth and MRCA like this:

```python
def path_length(tree, u, v):
    mrca = tree.mrca(u, v)
    if mrca == -1:
        return math.inf
    return tree.depth(u) + tree.depth(v) - 2 * tree.depth(mrca)
```

@jeremyguez, do you agree with this definition?

If we agree with this definition we can implement in C based on the definition of [get_mrca](https://github.com/tskit-dev/tskit/blob/main/c/tskit/trees.c#L3535). The first step would be implement in Python and get some tests in place within the ``test_tree_balance.py`` file added in #2246

**Comment by jeromekelleher** [comment]
For motivation, see #2245

**Comment by jeremyguez** [comment]
@jeromekelleher 
You mean `test_balance_metrics.py`?

I was just wondering: in that case the definition used in `test_balance_metrics.py` and the actual method in `trees.py` will be the same, unlike for #2246 where those were different (`sackin_index_definition(tree)` vs `sackin_index(self)`)?

**Comment by jeromekelleher** [comment]
It may not be worth implementing this in C btw. I doubt there'll be much difference between the implementation above and an optimised C version in most cases, and we probably won't end up using the function explicitly in the B2 metric (as we'd hope to keep the required state on the stack, rather than recompute lengths over and over).

So, I think we can close this once we have the version above in the Python API, documented and tested with some well chosen examples.

**Comment by jeremyguez** [comment]
@jeromekelleher I gave a first try at it at #2259.

**Comment by jeromekelleher** [comment]
Closed in #2259
